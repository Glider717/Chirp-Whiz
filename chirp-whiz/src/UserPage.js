import React from 'react';
import birdList from './birdList';
import 'bootstrap/dist/css/bootstrap.min.css';
import Table from 'react-bootstrap/Table';

export function winPercentage(index, correct, incorrect) {
    let percent = 0;

    if (correct[index] > 0 || incorrect[index] > 0)
        percent = (correct[index] / (correct[index] + incorrect[index]))*100;

    return percent.toFixed(2);
}

function UserPage(props) {

    return(
        <div>
            <h1>User Profile</h1><br />
            <h4>Username: {props.name}</h4>
            <h4>Email: {props.email}</h4><br /><br />
            <h2>Quiz Statistics:</h2><br />
            <Table striped bordered hover size="sm">
                <thead>
                    <tr>
                        <th>Bird</th>
                        <th style={{width: '300px'}}>Number of times<br /><span style={{color: 'green'}}>correctly</span> identified</th>
                        <th style={{width: '300px'}}>Number of times<br /><span style={{color: 'red'}}>incorrectly</span> identified</th>
                        <th style={{width: '300px'}}>Accuracy</th>
                    </tr>
                </thead>
                <tbody>
                    {birdList.map((bird, i) =>
                        <tr key={i}>
                            <th>{bird.name}</th>
                            <th>{props.correct[i]}</th>
                            <th>{props.incorrect[i]}</th>
                            <th>{winPercentage(i, props.correct, props.incorrect)}%</th>
                        </tr>
                    )}
                </tbody>
            </Table>
        </div>
    );
}

export default UserPage;