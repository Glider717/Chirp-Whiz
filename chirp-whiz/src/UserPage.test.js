import React from 'react';
import UserPage from './UserPage';
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent, waitForElement } from '@testing-library/react'

it('test UserPage renders', () => {
    const container = render(<UserPage 
        name='name'
        email='email'
        correct={[0,0,0]}
        incorrect={[0,0,0]}
      />)
    expect(container.firstChild).toMatchSnapshot();
});