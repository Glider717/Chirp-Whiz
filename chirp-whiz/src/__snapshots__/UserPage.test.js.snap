// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UserPage renders 1`] = `undefined`;
